### ISSUE TITLE
check_snmp: performance data includes warning/critical thresholds wrong w/ multiple OIDS

### ISSUE THREAD

**Issue by Tontonitch** [nagios-plugins/nagios-plugins#291]
```
[root@xxxxxxxxx ~]# /usr/lib64/nagios/plugins/check_snmp -V
check_snmp v2.1.4 (nagios-plugins 2.1.4)
[root@xxxxxxxxx ~]#
```

Refering to the NEWS:

> 2.1.2 2016-08-01
> 	ENHANCEMENTS
> 	check_snmp's performance data now also includes warning/critical
> 	  thresholds
> 

I tried that, since I am waiting for the feature.
```
[root@xxxxxxxxx ~]# /usr/lib64/nagios/plugins/check_snmp -P 2c -C xxxxxxxxx -H xxxxxxxxx -o "stConnectionsBlocked.0,stConnectionsLegitimate.0" --rate -w 1:,100: -c 10:,1000:
SNMP RATE CRITICAL - *0.009795797207* *2.264746262* | MCAFEE-MWG-MIB::stConnectionsBlocked.0=0.009795797207;1:,100:;10:,1000:;0;0; MCAFEE-MWG-MIB::stConnectionsLegitimate.0=2.264746262;1:,100:;10:,1000:;0;0;
[root@xxxxxxxxx ~]#
```

The warning/critical values included in the perfdata are the complete strings of -w / -c - not just the (parsed?) value used for the check against it.

For the sake of understanding correctly, I expected as result:
```
SNMP RATE CRITICAL - *0.009795797207* *2.264746262* | MCAFEE-MWG-MIB::stConnectionsBlocked.0=0.009795797207;1:;10:;0;0; MCAFEE-MWG-MIB::stConnectionsLegitimate.0=2.264746262;100:;1000:;0;0;
```

**Comment by hedenface** [comment]
Well - in 2.2.1 this was altered slightly already - such that the maximum value of the warnings/criticals are displayed - but that isn't quite right either.

**Comment by hedenface** [comment]
Fixed here: f838eaae4525054b3f339e2f1942327f2ce8f9b0

Also I cleaned the code up so the folks at home can follow along in the perfdata section of check_snmp.

Also, added a warning_string and critical_string to the threshold struct so that the warning and critical strings that were parsed are always saved with the threshold so that they can be recalled at a later time without any additional re-parsing. This will be useful for other plugins, and will drastically cut down on dev time.

Thanks!
